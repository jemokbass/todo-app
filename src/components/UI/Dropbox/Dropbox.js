import React, { useContext } from 'react';
import Dropzone from 'react-dropzone';
import { Controller } from 'react-hook-form';

import { LanguageContext } from '@src/shared/context';

const Dropbox = ({ control, name, errors, errorsMessage }) => {
  const resources = useContext(LanguageContext);

  return (
    <Controller
      control={control}
      name={name}
      defaultValue={[]}
      render={({ field: { onChange, onBlur, value } }) => (
        <>
          <Dropzone onDrop={onChange} maxFiles={1} maxSize={100000}>
            {({ getRootProps, getInputProps }) => (
              <div className="dropbox" {...getRootProps()}>
                <input {...getInputProps()} name={name} onBlur={onBlur} />
                <p>{resources.dropbox_desc}</p>
              </div>
            )}
          </Dropzone>
          <ul className="dropbox-list">
            {value?.map((file, index) => (
              <li className="dropbox-item" key={index}>
                <span className="dropbox-item__title">{file.name}</span>
                <span className="dropbox-item__size">{file.size} bytes</span>
              </li>
            ))}
          </ul>
          {errors && <span className="input-field__error">{errorsMessage}</span>}
        </>
      )}
    />
  );
};

export default Dropbox;
