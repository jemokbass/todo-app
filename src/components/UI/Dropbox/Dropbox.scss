.dropbox {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 80px;
  background: $element-color;
  border: 2px solid $element-color-darkest;
  border-radius: 10px;
  cursor: pointer;
}

.dropbox-item {
  display: block;

  &__title {
    font-size: 20px;
    color: $text-color-lightest;
  }

  &__size {
    margin: 0 0 0 20px;
  }
}
