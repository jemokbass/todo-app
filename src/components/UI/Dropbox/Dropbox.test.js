import React from 'react';
import { render } from '@testing-library/react';
import { useForm } from 'react-hook-form';

import Dropbox from './Dropbox';

const MockDropbox = props => {
  const { control } = useForm();

  return <Dropbox control={control} name="mock" {...props} />;
};

describe('Dropbox UI component', () => {
  it('Should render dropbox component', () => {
    render(<MockDropbox />);
  });

  describe('Should render snapshots', () => {
    it('Should render empty dropbox snapshot', () => {
      const { asFragment } = render(<MockDropbox />);
      expect(asFragment(<MockDropbox />)).toMatchSnapshot();
    });
  });
});
