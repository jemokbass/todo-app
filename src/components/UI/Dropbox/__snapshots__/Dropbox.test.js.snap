// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropbox UI component Should render snapshots Should render empty dropbox snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropbox"
    tabindex="0"
  >
    <input
      autocomplete="off"
      multiple=""
      name="mock"
      style="display: none;"
      tabindex="-1"
      type="file"
    />
    <p />
  </div>
  <ul
    class="dropbox-list"
  />
</DocumentFragment>
`;
